lib/vmdb/settings.rb

```diff
@@ -68,6 +73,18 @@ def self.save!(resource, hash)
       apply_settings_changes(resource, deltas)
     end
 
+    def self.save_yaml!(resource, contents)
+      require 'yaml'
+      hash =
+        begin
+          decrypt_passwords!(YAML.load(contents))
+        rescue => err
```

Comment:
If you look at the [previous code](https://github.com/ManageIQ/manageiq/pull/17494/files#diff-04cf6aae9623f8a3107672e3715bfa4fL71), this enumerated both StandardError and Psych::SyntaxError.  We had done this in the past because Psych::SyntaxError used to not derive from StandardError, but in Ruby 2.3 and 2.4 that is no longer the case, so we can just catch StandardError like normal.  I did this because rubocop rightly complains with [Lint/ShadowedException](https://www.rubydoc.info/gems/rubocop/0.52.1/RuboCop/Cop/Lint/ShadowedException)

```ruby
[1] pry(main)> Psych::SyntaxError < StandardError
=> true
[2] pry(main)> RUBY_VERSION
=> "2.3.3"
[3] pry(main)> Psych::VERSION
=> "2.1.0"
```